cost/aws/old_snapshots/aws_delete_old_snapshots.pt

```diff
@@ -229,15 +299,43 @@ escalation "esc_delete_snapshot" do
       description "Expain why are you approving the action"
     end
   end
-  run "delete_snapshot",data
+  run "take_action",data
 end
 
 ###############################################################################
 # Cloud Workflow
 ###############################################################################
 
- define delete_snapshot($data) do
- foreach $item in $data do
+define take_action($data) return $all_responses do
+  $all_responses = []
+  foreach $item in $data do
+    call deregisterImageFromSnapshot($item)
+    call delete_snapshot($item)
+  end
+end
+
+define deregisterImageFromSnapshot($item) do
+  foreach $image in $item["imageId"] do
+    $deregister_response = http_request(
+      auth: $$auth_aws,
+      https: true,
+      verb: "get",
+      host: "ec2."+$item["region"]+".amazonaws.com",
+      href: "/",
+      query_strings: {
+      "Action": "DeregisterImage",
+      "Version": "2016-11-15",
+      "ImageId": $image
+      }
+   )
+  $deregisterResponseResult = $deregister_response["code"]
+    if $deregisterResponseResult != 200
+      call sys_log("Inside deregisterImageFromSnapshot  defination", to_s($deregister_response))
```

Comment:
I like that you are logging the deregistration, consider reducing the summary.  `Deregister image snapshot` ?

Reply:
Done !